pkg/cmd/get/get_build_logs.go

```diff
@@ -322,90 +322,98 @@ func (o *GetBuildLogsOptions) getProwBuildLog(kubeClient kubernetes.Interface, t
 			return err
 		}
 	}
-	if build == nil {
+	if len(pipelineMap[name]) == 0 {
 		return fmt.Errorf("No Pipeline found for name %s in values: %s", name, strings.Join(names, ", "))
 	}
 
 	if tektonEnabled {
-		pri := build.(*tekton.PipelineRunInfo)
-		log.Logger().Infof("Build logs for %s", util.ColorInfo(name+suffix))
-		for _, stage := range pri.GetOrderedTaskStages() {
-			pr, err := tektonClient.TektonV1alpha1().PipelineRuns(ns).Get(pri.PipelineRun, metav1.GetOptions{})
-			if err != nil {
-				return err
-			}
-			if stage.Pod == nil {
-				// The stage's pod hasn't been created yet, so let's wait a bit.
-				f := func() error {
-					selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: map[string]string{
-						pipeline.GroupName + pipeline.PipelineRunLabelKey: pri.PipelineRun,
-					}})
-					if err != nil {
-						return err
+		var pipelines []*tekton.PipelineRunInfo
```

Comment:
If it were me I would for sure extract some functions here. This is ways too nested for my taste.